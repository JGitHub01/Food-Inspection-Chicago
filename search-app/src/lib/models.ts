export interface IResultItem {
  inspection_date: string
  dba_name: string
  address: string
  risk: string
}

export interface ISearchResults {
  keyword?: string
  items: IResultItem[]
  success?: boolean
  msg?: string
}
