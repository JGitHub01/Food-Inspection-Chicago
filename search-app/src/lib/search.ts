import {ISearchResults} from 'lib/models';

export const inspectionDataUrl = 'https://data.cityofchicago.org/resource/4ijn-s7e5.json';
export default async function search(keyword: string): Promise<ISearchResults> {
  const queryStr = new URLSearchParams({
    $q: keyword,
    $order: 'inspection_date DESC'
  }).toString();
  const searchUrl = `${inspectionDataUrl}?${queryStr}`;
  try {
    const response = await fetch(searchUrl, {
      method: 'GET',
      headers: {
        "Accept": "application/json"
      }
    });
    if (response.status === 200) {
      return {
        keyword: keyword,
        items: await response.json(),
        success: true
      };
    } else {
      return {
        items: [],
        success: false,
        msg: response.statusText
      }
    }
  } catch (error) {
    return {
      items: [],
      success: false,
      msg: error.toString()
    }
  }
}
