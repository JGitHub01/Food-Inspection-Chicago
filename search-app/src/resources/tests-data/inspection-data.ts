export const inspection_data =
[
    {
        "inspection_id": "2509677",
        "dba_name": "SCHUBA'S TAVERN/TIED HOUSE",
        "aka_name": "SCHUBA'S TAVERN",
        "license_": "2549711",
        "facility_type": "Restaurant",
        "risk": "Risk 1 (High)",
        "address": "3153-3159 N SOUTHPORT AVE ",
        "city": "CHICAGO",
        "state": "IL",
        "zip": "60657",
        "inspection_date": "2021-05-13T00:00:00.000",
        "inspection_type": "Canvass Re-Inspection",
    },
    {
        "inspection_id": "2509457",
        "dba_name": "KENTUCKY FRIED CHICKEN",
        "aka_name": "KENTUCKY FRIED CHICKEN",
        "license_": "2565166",
        "facility_type": "Restaurant",
        "risk": "Risk 1 (High)",
        "address": "3927 N HARLEM AVE ",
        "city": "CHICAGO",
        "state": "IL",
        "zip": "60634",
        "inspection_date": "2021-05-10T00:00:00.000",
        "inspection_type": "Complaint",
    }
  ]
