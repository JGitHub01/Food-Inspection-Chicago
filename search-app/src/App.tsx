import { useState } from 'react';
import SearchBar from 'view/SearchBar/SearchBar';
import SearchResults from 'view/SearchResults/SearchResults';
import search from 'lib/search';
import './styles/app.scss';

function App() {
  const [keyword, setKeyword] = useState<string>("");
  const [results, doSearch] = useSearch();
  return (
    <div className="search-app-view">
      <SearchBar
        keyword={keyword}
        onKeywordChange={setKeyword}
        onSearch={doSearch}
      />
      <SearchResults results={results}/>
    </div>
  );
}

const useSearch = (): [string, (keyword: string) => void] => {
  const [results, setResults] = useState<string>("Search results for: ");
  const doSearch = (keyword: string) => {
    const results = search(keyword);
    setResults(results);
  }

  return [results, doSearch];
}
export default App;
