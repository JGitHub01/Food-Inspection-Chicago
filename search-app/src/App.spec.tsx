import {render, waitFor} from '@testing-library/react';
import {rest} from 'msw';
import { setupServer } from 'msw/node';
import userEvent from '@testing-library/user-event';
import {inspectionDataUrl} from 'lib/search';
import { inspection_data} from 'resources/tests-data/inspection-data';
import App from './App';

const server = setupServer(
  rest.get(inspectionDataUrl, (req, res, ctx) => {
    return res(ctx.json(inspection_data));
  })
);

describe('App', () => {
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());

  it('Should render the basic snapshot view', () => {
    const {container} = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('Should initiate a search and display search result when the search button is clicked', async () => {
    const {getByTestId, getAllByTestId} = render(<App />);
    const searchField = getByTestId('search-field');
    const searchButton = getByTestId('search-button');
    userEvent.type(searchField, 'fried');
    userEvent.click(searchButton);
    await waitFor(() => getAllByTestId('result-item'));
    expect(getAllByTestId('result-item')).toHaveLength(2);
  });
});
