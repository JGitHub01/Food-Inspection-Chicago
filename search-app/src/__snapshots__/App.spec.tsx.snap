// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render the basic snapshot view 1`] = `
<div>
  <div
    class="search-app__sticky-header"
  >
    <h2
      class="title"
    >
      Chicago Food Inspection
    </h2>
    <div
      class="search-bar"
      data-testid="search-bar"
    >
      <div
        class="MuiFormControl-root MuiTextField-root search-bar__text"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
          data-shrink="false"
        >
          Type a keyword
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input"
            data-testid="search-field"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="search-bar__button"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="search-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Search
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="search-results"
    data-testid="search-results"
  >
    <div
      class="search-results__not-found"
    >
      Please enter a keyword
    </div>
  </div>
</div>
`;
